### PR TITLE
fix: preserve page when go back from resource show page (#1555)

### PR DESCRIPTION
**Pull request by dmitrytrager**
# Description

Preserve page number when going back from resource show page.
Page number is fetched from referral url by parsing its query params.

Another approach can be using pagination data from index page as a parameter for generated links and then just checking `params` for `back_path`. The problem with this option is not that we will have to pass `@pagy` down through components tree, but that page number will be present in show links as well, which seems redundant.

Fixes [#1555](https://github.com/orgs/avo-hq/projects/8?pane=issue&itemId=47768610)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [x] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Run local dummy app
2. Open project list and go to page 2
3. Select some project and open show page
4. Click 'Go back'
5. Make sure you are on page 2 

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Comment by dmitrytrager** [comment]
Updated code style here a bit

**Review by Paul-Bob — CHANGES_REQUESTED** [review]
Thanks for the initiative @dmitrytrager is a great beginning!

Let's refactor it a little bit and apply the same logic when editing.

I was thinking about a private method inside `Avo::ResourceComponent` class:
```ruby
  def referrer_params
    {
      page: referrer_params["page"]
    }.compact
  end
```

This way we can apply it to all needed paths like `**referrer_params`. Since we compact the `referrer_params` that means `nil` values will be erased from the hash and we do no longer need the `via_page?` method and verification.

We can now apply this strategy:
```ruby
# app/components/avo/views/resource_edit_component.rb
# Line 26
return helpers.resource_path(record: @resource.record, resource: @resource, **referer_params)

# Line 33
helpers.resources_path(resource: @resource, **referer_params)
```

Inside `app/components/avo/views/resource_show_component.rb` file we can replace 

```ruby
via_page? ?
        helpers.resources_path(resource: @resource, page: referrer_params["page"]) :
        helpers.resources_path(resource: @resource)
```

By:

```ruby
helpers.resources_path(resource: @resource, **referer_params)
```

Thanks again for looking into it!

**Review by Paul-Bob — APPROVED** [review]
Thank you @dmitrytrager !

File: app/components/avo/resource_component.rb [hunk 1]
```diff
@@ -116,6 +117,10 @@ def via_resource?
     (params[:via_resource_class].present? || params[:via_relation_class].present?) && params[:via_record_id].present?
   end
 
+  def keep_referrer_params
+    { page: referrer_params["page"] }.compact
```
Comment (adrianthedev) [inline app/components/avo/resource_component.rb:121]:
```suggestion
    {page: referrer_params["page"]}.compact
```

**Review by adrianthedev — APPROVED** [review]

**Pull request merged**